include/netlink/route/nexthop.h

```diff
@@ -61,11 +61,23 @@ extern int		rtnl_route_nh_str2flags(const char *);
 /*
  * nexthop encapsulations
  */
+extern int		rtnl_route_nh_get_encap_type(struct rtnl_nexthop *nh);
+
 extern int		rtnl_route_nh_encap_mpls(struct rtnl_nexthop *nh,
 						 struct nl_addr *addr,
 						 uint8_t ttl);
 extern struct nl_addr *	rtnl_route_nh_get_encap_mpls_dst(struct rtnl_nexthop *);
 extern uint8_t		rtnl_route_nh_get_encap_mpls_ttl(struct rtnl_nexthop *);
+
+
+extern int		rtnl_route_nh_get_encap_seg6_mode(struct rtnl_nexthop * );
+extern int		rtnl_route_nh_get_encap_seg6_srh(struct rtnl_nexthop *, void **);
```

Comment:
the `void**` here is really the `struct ipv6_sr_hdr`. OK, I think we can do this, but then it should be
```
struct ipv6_sr_hdr;
extern int		rtnl_route_nh_get_encap_seg6_srh(struct rtnl_nexthop *, const struct ipv6_sr_hdr **);
```

also, the user wouldn't know the actual size (`IPV6_EXTHDR_LEN`), so the size should be returned by `rtnl_route_nh_get_encap_seg6_srh()`.


The alternative (maybe better) would be to add a `struct rtnl_ipv6_sr_hdr` that is fully under libnl control, and which possibly exposes a more convenient layout of fields (e.g. the number of segments as a fields of the struct).

Reply:
done